system-tests/resources/cli-tools/entrypoint.sh

```diff
@@ -3,34 +3,92 @@
+          onboardingCompleted=true
+          break
+      else
+          echo "Onboarding is not completed yet for $participantName. Waiting for 1 seconds..."
+          sleep 1
+      fi
+      retryCount=$((retryCount+1))
+  done
+  if [ "$onboardingCompleted" == false ]; then
+      echo "$participantDid onboarding is not completed yet. Exiting..."
+      exit 1
+  fi
+}
+
+function registerParticipant() {
+  local participantName="$1"
+  local participantDid="$2"
+
+  echo "Registering $participantName"
+  java -jar registration-service-cli.jar \
+              -d="did:web:did-server:registration-service" \
+              --http-scheme \
+              -k=/resources/vault/$participantName/private-key.pem \
+              -c="$participantDid" \
+               participants add
+}
+
+function seedVerifiedCredentials() {
+  local participantName="$1"
+  local participantDid="$2"
+  local region="$3"
+
+   for subject in '"region": "'$region'"' '"gaiaXMember": "true"'
+   do
+     echo "Seeding VC for $participantName: $subject"
+     vcId=$(uuidgen)
+     java -jar identity-hub-cli.jar \
+                 -s="http://$participantName:8181/api/identity-hub" \
+                 vc add \
+                 -c='{"id":"'$vcId'","credentialSubject":{'"$subject"'}}' \
+                -b="$participantDid" \
+                 -i="did:web:did-server:gaia-x" \
+                 -k="/resources/vault/gaia-x/private-key.pem"
+   done
+}
+
 PARTICIPANTS=(company1:eu company2:eu company3:us)
 
-# Register dataspace participants
 for participant in "${PARTICIPANTS[@]}"; do
     participantArray=(${participant//:/ })
 
     participantName=${participantArray[0]}
     region=${participantArray[1]}
     participantDid="did:web:did-server:$participantName"
 
-    for subject in '"region": "'$region'"' '"gaiaXMember": "true"'
-    do
-         echo "Seeding VC for $participantName: $subject"
-         vcId=$(uuidgen)
-         java -jar identity-hub-cli.jar \
-                -s="http://$participantName:8181/api/identity-hub" \
-                 vc add \
-                -c='{"id":"'$vcId'","credentialSubject":{'"$subject"}'}' \
-                -b="$participantDid" \
-                -i="did:web:did-server:gaia-x" \
-                -k="/resources/vault/gaia-x/private-key.pem"
-    done
-
-    echo "Registering $participantName"
-    java -jar registration-service-cli.jar \
-                -d="did:web:did-server:registration-service" \
-                --http-scheme \
-                -k=/resources/vault/$participantName/private-key.pem \
-                -c="$participantDid" \
-                 participants add
+  # seed  vc for participant
+  seedVerifiedCredentials "$participantName" "$participantDid" "$region"
+
+  # Register dataspace participants
+  registerParticipant "$participantName" "$participantDid"
+
+  # Wait for participant registration.
+  waitForOnboarding "$participantName" "$participantDid"
```

Comment:
seems inefficient to wait in a loop like that. I'm open to a discussion on this. What's the time impact of this?

Would an alternative be to (1) rearrange the script to trigger all onboardings first and then wait for all to complete, or (2) run this container three times?

Reply:
As discussed with @algattik, going for option 1 and separated seeding/registration from waiting.